asset-transfer-events/application-gateway-typescript/src/app.ts

```diff
@@ -0,0 +1,365 @@
+ */
+function checkAsset(mspId:string, asset:{
+    ID: string,
+    Color: string,
+    Size: string,
+    Owner: string,
+    AppraisedValue: string,
+    asset_properties:{
+        object_type: string,
+		asset_id: string,
+		Price: string,
+		salt: string
+    }
+}, color:string, size:string, owner:string, appraisedValue:string, price?:string) {
+    console.log(`<-- Query results from ${mspId}`);
+
+    console.log(`*** verify asset ${asset.ID}`);
+
+    if (asset) {
+        if (asset.Color === color) {
+            console.log(`*** asset ${asset.ID} has color ${asset.Color}`);
+        } else {
+            console.log(`*** asset ${asset.ID} has color of ${asset.Color}`);
+        }
+        if (asset.Size === size) {
+            console.log(`*** asset ${asset.ID} has size ${asset.Size}`);
+        } else {
+            console.log(`*** Failed size check from ${mspId} - asset ${asset.ID} has size of ${asset.Size}`);
+        }
+        if (asset.Owner === owner) {
+            console.log(`*** asset ${asset.ID} owned by ${asset.Owner}`);
+        } else {
+            console.log(`*** Failed owner check from ${mspId} - asset ${asset.ID} owned by ${asset.Owner}`);
+        }
+        if (asset.AppraisedValue === appraisedValue) {
+            console.log(`*** asset ${asset.ID} has appraised value ${asset.AppraisedValue}`);
+        } else {
+            console.log(`*** Failed appraised value check from ${mspId} - asset ${asset.ID} has appraised value of ${asset.AppraisedValue}`);
+        }
+        if (price) {
+            if (asset.asset_properties && asset.asset_properties.Price === price) {
+                console.log(`*** asset ${asset.ID} has price ${asset.asset_properties.Price}`);
+            } else {
+                console.log(`*** Failed price check from ${mspId} - asset ${asset.ID} has price of ${asset.asset_properties.Price}`);
+            }
+        }
+    }
+}
+
+/**
+ *  Submit transaction, blocking until the transaction has been committed on the ledger.
+    The 'transient' data will not get written to the ledger, and is used to send sensitive data to the trusted endorsing peers.
+    The gateway will only send this to peers that are included in the ownership policy of all collections accessed by the chaincode function.
+ */
+async function createAssetPrivate(contract: Contract): Promise<void> {
+    console.log('\n--> Submit Transaction: CreateAsset, creates new asset with ID, Color, Size, Owner and AppraisedValue arguments');
+
+    // create the private data with salt and assign to the transaction
+    const asset_properties = {
+        object_type: 'asset_properties',
+        asset_id: assetId,
+        Price: '90',
+        salt: Buffer.from(Date.now().toString()).toString('hex')
+    };
+
+    await contract.submit(
+        'CreateAsset',
+        {   arguments:[assetId, 'blue', '10', 'James', '100'],
+            transientData: asset_properties,
+            endorsingOrganizations: [mspId]
```

Comment:
You shouldn't need to specify the endorsing orgs in this case. The gateway will select its own org by default.

Reply:
Without the endorsement orgs,I got 'channel shut down' error